File: Rating 800/1385B.RestoreThePermutationByMerger.cpp
// https://codeforces.com/problemset/problem/1385/B
// 1385 B. Restore the Permutation by Merger
// File Creation Date: 24-Oct-2020
// Author: Gourav(https://github.com/GouravKhunger)

#include <bits/stdc++.h>
using namespace std;

int main() {
  int t;
  cin>>t;
  for(int i=0;i<t;i++) {
    int n;
    cin>>n;
    int a[2*n];
    vector<int> v;
    for(int j=0;j<2*n;j++) {
      cin>>a[j];
    }
    for(int j=0;j<2*n;j++) {
      auto pos = find(v.begin(), v.end(), a[j]);
      if(pos==v.end()) {
        v.push_back(a[j]);
      }
    }
    for(auto x : v) {
      cout<<x<<" ";
    }
    cout<<"\n";
  }
  return 0;
}

File: Rating 800/1385B.RestoreThePermutationByMerger.py
'''
    https://codeforces.com/contest/1385/problem/B
    1385 B. Restore the Permutation by Merger
    File Creation Date: 24-Oct-2020
    Author: Biraj(https://github.com/BirajCoder)
'''

for _ in range(int(input())):
    n=int(input())
    li=list(map(int,input().split()))
    if len(li)==1:
        print(0)
    else:
        max1=-1
        max2=-1
        index1=-1
        index2=-1
        for i in range(len(li)-1,0,-1):
            if li[i-1]<li[i]:
                max1=li[i]
                index1=i
                break
        i-=1
        while(i>0):
            if li[i-1]>li[i]:
                max2=li[i-1]
                index2=i-1
                break
            i-=1
        if index2 == -1 and index1 == -1:
            print(0)
        else:
            print(i)